packages/backend/app/pkg/kvs/client.go

```diff
@@ -0,0 +1,63 @@
+package kvs
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/stlatica/stlatica/packages/backend/app/pkg/logger"
+)
+
+// Client is an interface for key-value store.
+type Client interface {
+	// SetValue sets a key value pair in valkey.
+	SetValue(ctx context.Context, key string, value string) error
+	// GetValue retrieves a value for a given key from valkey.
+	GetValue(ctx context.Context, key string) (string, error)
+}
+
+type client struct {
+	client    *redis.Client
+	appLogger *logger.AppLogger
+}
+
+// NewClient creates a new key-value store client.
+func NewClient(ctx context.Context, appLogger *logger.AppLogger, addr string,
+	password string, defaultDB int) Client {
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     addr,     // address of kvs server
+		Password: password, // password of kvs
+		DB:       defaultDB,
+	})
+
+	// pingを飛ばす
+	_, err := rdb.Ping(ctx).Result()
+	if err != nil {
+		log.Fatalf("Err: Could not connect to Redis: %v", err)
+		return nil
```

Comment:
アプリケーションの起動時に出るエラーは基本的にpanicで終了させるように統一したいのでpanicに変更してもらえますか？

https://zenn.dev/snowcrush/articles/21f28163e067cb

```suggestion
		panic(fmt.Sprintf("Err: Could not connect to Redis: %v", err))
```

---

ただ本当はここでpanicを起こすのではなく`cmd` packageの方で起こした方が良さそうですね 🤔 
この部分のルールを決めていなかったので後でscrapboxの方に明記しておきます。他の箇所も合わせて修正するので今回はここでpanicを直接起こしてもらって大丈夫です。